skimage/metrics/tests/test_overlap.py

```diff
@@ -0,0 +1,59 @@
+from skimage.metrics.overlap import (
+        Rectangle,
+        intersect,
+        intersection_over_union,
+        )
+from skimage._shared import testing
+
+height1, width1 = 2, 4
+rectangle1 = Rectangle((0, 0), dimensions=(height1, width1))
+
+height2 = width2 = 3
+rectangle2 = Rectangle((2, 3), dimensions=(height2, width2))
+
+rectangle3 = Rectangle((0, 0), bottom_right=(2, 2))
+
+rectangle4 = Rectangle((10, 10), dimensions=(5, 5))
+
+rect_inter13 = intersect(rectangle1, rectangle3)
+
+
+def test_area():
+    assert rectangle1.area == height1 * width1
+    assert rectangle2.area == height2 * width2
+    assert rectangle3.area == 2 * 2
+
+
+def test_constructor_dimensions():
+    assert tuple(rectangle1.top_left) == (0, 0)
+    assert tuple(rectangle1.bottom_right) == (height1, width1)
+    with testing.raises(ValueError):
+        Rectangle((0, 0), dimensions=(-5, 5))
+
+def test_constructor_bottom_corner():
+    assert tuple(rectangle3.top_left) == (0, 0)
+    assert tuple(rectangle3.dimensions) == (2, 2)
+    with testing.raises(ValueError):
+        Rectangle((2,2), bottom_right=(0, 0))
+
+def test_intersection():
+    assert not intersect(rectangle1, rectangle2)
+    assert intersect(rectangle1, rectangle3)
```

Comment:
Shouldn't we test for the intersection value? I suggest also to test for the case where `rectangle1` in fully included in `rectangle2` and for every case that intersection is symmetric (`intersect(r1, r2) == intersect(r2, r1)`).

Reply:
We should also test for the edge cases:
- rectangles only share a corner
- rectangles share an edge.